### PR TITLE
Gh page v2

### PR DESCRIPTION
**Pull request by theodoreb**
### Description of the Change

Updated design and presentation of the ovh-ux page using vuejs and github API.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+
+[{package,bower}.json]
+indent_style = space
+indent_size = 2
```
Comment (antleblanc) [inline .editorconfig]: insert_final_newline on this file :)

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            "property": "pow"
+        }],
+        "no-magic-numbers": 0,
+        "no-underscore-dangle": 0,
+        "prefer-arrow-callback": 0,
+        "prefer-template": 0,
+        "object-shorthand": 0,
+        "class-methods-use-this": 0
+    },
+    "globals": {
```
Comment (antleblanc) [inline .eslintrc.js]: Are you sure about those globals?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
-# OVH-UX.github.io
+# ovh-ux-github-page
 
-![github pages](https://github.com/ovh-ux/ovh-ux-guidelines/blob/master/media/guidelinesBanner.png)
+> A Vue.js project
```
Comment (antleblanc) [inline README.md]: Add some description to tell more about this repository.

File: index.html [hunk 4]
```diff
@@ -1,16 +1,12 @@
...
-            </div>
-        </section>
-    {% endif %}
-{% endfor %}
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, shrink-to-fit=no">
+    <title>ovh-ux-github-page</title>
```
Comment (antleblanc) [inline index.html]: Be more descriptive on the `<title>` element.

File: package.json [hunk 5]
```diff
@@ -1,31 +1,80 @@
 {
-  "name": "ovh-ux.github.io",
-  "version": "0.0.1",
-  "description": "ovh-ux github pages",
-  "main": "index.js",
+  "name": "ovh-ux-github-page",
+  "version": "1.0.0",
+  "description": "A Vue.js project",
+  "author": "Camille Gicquel <camille.gicquel@corp.ovh.com>",
```
Comment (antleblanc) [inline package.json]: `OVH SAS`

File: package.json [hunk 6]
```diff
@@ -1,31 +1,80 @@
...
+  "scripts": {
+    "dev": "webpack-dev-server --inline --progress --config build/webpack.dev.conf.js",
+    "start": "npm run dev",
+    "build": "node build/build.js"
+  },
   "dependencies": {
-    "cz-ovh-commit": "^1.0.0"
+    "axios": "^0.18.0",
+    "babel-eslint": "^8.2.2",
+    "eslint": "^4.19.1",
```
Comment (antleblanc) [inline package.json]: Reorder packages as devDependencies or not such as `babel`, `eslint`, `node-sass` for instance.

File: src/App.vue [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  methods: mapActions({
+    upNbrRepos: 'nbrRepos',
+    upLastRepo: 'lastPushed',
+    upAboutTechno: 'aboutTechno',
+    upManager: 'manager',
+    upOtherProjects: 'otherProjects'
+  }),
+  mounted: function() {
+    this.upNbrRepos(),
```
Comment (antleblanc) [inline src/App.vue]: Enhance naming functions (proposal):
- `upNbrRepos` => `updateNumberOfPublicRepositories`
- `nbrRepos` => `getNumbersOfPublicRepositories` (count)

File: src/store/Github/actions.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
+import { api } from "../../utils/request";
+import lodash from "lodash";
```
Comment (antleblanc) [inline src/store/Github/actions.js]: missing `as`? (or you just want to use `sortBy` and `pick`?)

File: src/store/Github/state.js [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+    mainNavItems: [
+        "Home",
+        "About",
+        "Works",
+        "Contact"
+    ],
+    aboutRecentItems: [
+        {
+            icon: "puzzle",
+            property: "repositories on github",
```
Comment (antleblanc) [inline src/store/Github/state.js]: `github` => `GitHub`.

File: src/store/Github/state.js [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+                {
+                    property: "Contributors",
+                    icon: "person",
+                    value: 0
+                },
+                {
+                    property: "since last update",
+                    icon: "clock",
+                    value: "0 mn"
+                }
```
Comment (antleblanc) [inline src/store/Github/state.js]: Stars would be great 🌟

File: src/store/Github/state.js [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+            link: "https://github.com/ovh-ux/ovh-manager-dedicated"
+        },
+        {
+            name: "Other projects",
+            type: "listing",
+            description: "Discover here some of the other projects, guidelines components... we have built.",
+            items: [
+                {
+                    name: "ovh-ui-kit-bs",
+                    description: "A bootstrap theme for the OVH managers, based on ovh-ui-kit.",
```
Comment (antleblanc) [inline src/store/Github/state.js]: `bootstrap` => `Bootstrap`.

**Comment by theodoreb** [comment]
Still missing fallback data to not have an empty page when we're over the github limit. Once that's done I'll merge.

File: src/store/Github/state.js (hunk 10 above)
Comment (theodoreb) [reply src/store/Github/state.js]: new feature, in the context of this PR it's scope creep.

File: .babelrc [hunk 12]
```diff
@@ -0,0 +1,12 @@
+{
+  "presets": [
+    ["env", {
+      "modules": false,
+      "targets": {
+        "browsers": ["> 1%", "last 2 versions", "not ie <= 8"]
```
Comment (FredericEspiau) [inline .babelrc]: Drop `last 2 versions`, it includes browsers nobody uses anymore (i.e. Opera Mobile)

**Comment by theodoreb** [comment]
master holds the compiled code and develop holds the dev code. PR not needed anymore.

**Pull request closed without merging**